prism/src/main/java/org/apache/falcon/resource/channel/HTTPChannel.java

```diff
@@ -170,8 +171,20 @@ private String pathValue(Method method, Object... args) throws FalconException {
                     String annotationClass = paramAnnotation.annotationType().getName();
 
                     if (annotationClass.equals(QueryParam.class.getName())) {
-                        queryString.append(getAnnotationValue(paramAnnotation, "value")).
-                                append('=').append(arg).append("&");
+                        if (args[index] instanceof List){
+                            List lifecycle = (List) args[index];
```

Comment:
I agree with Pavan and think this piece of code is ugly and not scalable, we have faced issues with it in the past. It needs to be generic enough to work with all sort of parameters but I understand that the way we are doing it right now (by reflection etc.) we will never be able to do it in a generic fashion and it might not even be possible to use it if someone uses advance features like custom beans etc. However, fixing it will be too much to accommodate in this JIRA, we can create another JIRA to address this issue.

:+1:  for this change from my side.